src/jit/function-builder.cc

```diff
@@ -76,6 +76,64 @@ inline Opcode ReadOpcodeAt(const uint8_t* pc) {
   return ReadOpcode(&pc);
 }
 
+#define CHECK_TRAP_IN_HELPER(...)                \
+  do {                                           \
+    wabt::interp::Result result = (__VA_ARGS__); \
+    if (result != wabt::interp::Result::Ok) {    \
+      return static_cast<Result_t>(result);      \
+    }                                            \
+  } while (0)
+
+FunctionBuilder::Result_t FunctionBuilder::CallHelper(wabt::interp::Thread* th, wabt::interp::IstreamOffset offset) {
+  // no need to check if JIT was enabled since we can only get here it was
+  auto meta_it = th->env_->jit_meta_.find(offset);
+
+  auto call_interp = [&](){
+    th->set_pc(offset);
+    auto last_jit_frame = th->last_jit_frame_;
+    th->last_jit_frame_ = th->pc_;
+    th->set_pc(offset);
+    wabt::interp::Result result = wabt::interp::Result::Ok;
+    while (result == wabt::interp::Result::Ok) {
+      result = th->Run(1000);
+    }
+    th->last_jit_frame_ = last_jit_frame;
+    return result;
+  };
+
+  auto currnt_pc = &(th->GetIstream()[th->pc_]);
```

Comment:
This is still pushing the wrong address because the thread's `pc_` value is stale at this point. Not only does it not get updated in the JITted function, the outer interpreter doesn't even save its value within interpreted code until it returns.

Reply:
*Sigh...* Not sure why I keep getting this one wrong :disappointed: .